pkg/distro/defs/loader.go

```diff
@@ -54,6 +55,45 @@ type imageType struct {
 	PartitionTables map[string]*disk.PartitionTable `yaml:"partition_table"`
 	// override specific aspects of the partition table
 	PartitionTablesOverrides *partitionTablesOverrides `yaml:"partition_table_override"`
+
+	ImageConfig imageConfig `yaml:"image_config,omitempty"`
+
+	Name          string
+	Filename      string
+	MimeType      string `yaml:"mime_type"`
+	Environment   string
+	KernelOptions []string `yaml:"kernel_options"`
+	Bootable      bool
+	DefaultSize   uint64 `yaml:"default_size"`
+	// the image func name: disk,container,live-installer,...
+	Image                  string
+	BuildPipelines         []string `yaml:"build_pipelines"`
+	PayloadPipelines       []string `yaml:"payload_pipelines"`
+	Exports                []string
+	RequiredPartitionSizes map[string]uint64 `yaml:"required_partition_sizes"`
+
+	Platforms map[string]ImagePlatform
```

Comment:
Fww, this is wrong, platforms must be a list instead of a map[arch]ImagePlatform as it is defined here. We could have mutliple aarch64 images for example with different platform config/boot files etc